services/gitdiff/gitdiff.go

```diff
@@ -955,6 +971,36 @@ parsingLoop:
 	return diff, nil
 }
 
+func skipToNextDiffHead(input *bufio.Reader) (line string, err error) {
+	// need to skip until the next cmdDiffHead
+	isFragment, wasFragment := false, false
+	var lineBytes []byte
+	for {
+		lineBytes, isFragment, err = input.ReadLine()
+		if err != nil {
+			return
+		}
+		if wasFragment {
+			wasFragment = isFragment
+			continue
+		}
+		if bytes.HasPrefix(lineBytes, []byte(cmdDiffHead)) {
```

Comment:
`[]byte(cmdDiffHead)` should this be a constant?

Reply:
Go will compile that away.